agent/engine/docker_task_engine.go

```diff
@@ -868,6 +878,13 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		return dockerapi.DockerContainerMetadata{Error: apierrors.NamedError(hcerr)}
 	}
 
+	// If LogRouter values are present, check to see if they are the ones expected and
+	// then specify the log driver options:  tag and fluentd-address are added
+	// For reference - https://docs.docker.com/config/containers/logging/fluentd/
+	if container.LogRouter != nil && container.LogRouter.Type == logConfigType {
```

Comment:
For a task using log router, there is one "log router container", and 1 or many "container that uses log router". The "log router container" has container.LogRouter != nil that contains definition for the log router, while a "container that uses log router" has a special log driver name "logrouter" (or whatever the name we end up using). The log driver name for a container is saved in hostConfig.LogConfig.Type. So you should check hostConfig.LogConfig.Type here, rather than container.LogRouter.Type.